pkg/appgateserver/cmd/cmd.go

```diff
@@ -0,0 +1,199 @@
+the address of the application that is sending the request. This is so that the server
+can generate the correct ring for the application and sign the request.`,
+		Args: cobra.NoArgs,
+		RunE: runAppGateServer,
+	}
+
+	cmd.Flags().StringVar(&signingKeyName, "signing-key-name", "", "The name of the key that will be used to sign relays")
+	cmd.Flags().StringVar(&listeningEndpoint, "listening-endpoint", "http://localhost:42069", "The host and port that the server will listen on")
+	cmd.Flags().StringVar(&cometWebsocketUrl, "comet-websocket-url", "ws://localhost:36657/websocket", "The URL of the tendermint websocket endpoint to interact with the chain")
+	cmd.Flags().BoolVar(&selfSigning, "self-signing", false, "Whether the server should sign all incoming requests with its own ring (for applications)")
+
+	cmd.Flags().String(flags.FlagKeyringBackend, "", "Select keyring's backend (os|file|kwallet|pass|test)")
+	cmd.Flags().String(flags.FlagNode, "tcp://localhost:36657", "tcp://<host>:<port> to tendermint rpc interface for this chain")
+
+	return cmd
+}
+
+func runAppGateServer(cmd *cobra.Command, _ []string) error {
+	// Create a context that is cancelled when the command is interrupted
+	ctx, cancelCtx := context.WithCancel(cmd.Context())
+
+	// Retrieve the client context for the chain interactions.
+	clientCtx := cosmosclient.GetClientContextFromCmd(cmd)
+
+	// Parse the listening endpoint.
+	listeningUrl, err := url.Parse(listeningEndpoint)
+	if err != nil {
+		cancelCtx()
+		return fmt.Errorf("failed to parse listening endpoint: %w", err)
+	}
+
+	// Obtain the tendermint websocket endpoint from the client context.
+	cometWSUrl, err := url.Parse(clientCtx.NodeURI + "/websocket")
+	if err != nil {
+		cancelCtx()
+		return fmt.Errorf("failed to parse block query URL: %w", err)
+	}
+	cometWSUrl.Scheme = "ws"
+	// If the comet websocket URL is not provided, use the one from the client context.
+	if cometWebsocketUrl == "" {
+		cometWebsocketUrl = cometWSUrl.String()
+	}
+
+	log.Printf("INFO: Creating block client, using websocket URL: %s...", cometWebsocketUrl)
+
+	// Create the block client with its dependency on the events client.
+	eventsQueryClient := eventsquery.NewEventsQueryClient(cometWebsocketUrl)
+	deps := depinject.Supply(eventsQueryClient)
+	blockClient, err := blockclient.NewBlockClient(ctx, deps, cometWebsocketUrl)
+	if err != nil {
+		cancelCtx()
+		return fmt.Errorf("failed to create block client: %w", err)
+	}
+
+	log.Println("INFO: Creating AppGate server...")
+
+	key, err := clientCtx.Keyring.Key(signingKeyName)
+	if err != nil {
+		cancelCtx()
+		return fmt.Errorf("failed to get key from keyring: %w", err)
+	}
+
+	appAddress, err := key.GetAddress()
+	if err != nil {
+		cancelCtx()
+		return fmt.Errorf("failed to get address from key: %w", err)
+	}
+	signingAddress := ""
+	if selfSigning {
+		signingAddress = appAddress.String()
+	}
+
+	signingKey, err := recordLocalToScalar(key.GetLocal())
```

Comment:
Can you add a comment on what `recordLocalToScalar` is doing above (just one sentence)